src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java

```diff
@@ -297,10 +304,19 @@ protected SCMRevision retrieve(SCMHead head, TaskListener listener)
         }
 
         if (head instanceof BitbucketBranchSCMHead) {
+            // TODO: JENKINS-73267 fetch commit from source rather than cached value
             return new BitbucketSCMRevision((BitbucketBranchSCMHead) head,
                     ((BitbucketBranchSCMHead) head).getLatestCommit());
         }
 
+        if (head instanceof BitbucketTagSCMHead) {
+            // This was previously a GitTagSCMHead and needs to be property retrieved
+            // Perform a fetch of the tag from the remote.
+            // Create a new BitbucketSCMRevision from the fetched tag.
+            BitbucketTag fetchedTag = fetchBitbucketTag((BitbucketTagSCMHead) head, listener);
+            return new BitbucketSCMRevision((BitbucketTagSCMHead) head, fetchedTag.getLatestCommit());
```

Comment:
If you go with my suggestion below, you can then change this to
```java
Optional<BitbucketTag> fetchedTag = fetchBitbucketTag((BitbucketTagSCMHead) head, listener);
            return new BitbucketSCMRevision((BitbucketTagSCMHead) head, fetchedTag.map(BitbucketTag::getLatestCommit).orElse(null));
```

of course deciding if `null` or empty String is the correct default.